server/service/osquery.go

```diff
@@ -1232,20 +1232,92 @@ func getFailingCalendarPolicies(policyResults map[uint]*bool, calendarPolicies [
 // We do this to not grow the main software queries and to ingest
 // all software together (one direct ingest function for all software).
 func preProcessSoftwareResults(
-	hostID uint,
+	host *fleet.Host,
 	results *fleet.OsqueryDistributedQueryResults,
 	statuses *map[string]fleet.OsqueryStatus,
 	messages *map[string]string,
 	overrides map[string]osquery_utils.DetailQuery,
 	logger log.Logger,
 ) {
+	//
 	vsCodeExtensionsExtraQuery := hostDetailQueryPrefix + "software_vscode_extensions"
-	preProcessSoftwareExtraResults(vsCodeExtensionsExtraQuery, hostID, results, statuses, messages, osquery_utils.DetailQuery{}, logger)
+	preProcessSoftwareExtraResults(vsCodeExtensionsExtraQuery, host.ID, results, statuses, messages, osquery_utils.DetailQuery{}, logger)
 
 	for name, query := range overrides {
 		fullQueryName := hostDetailQueryPrefix + "software_" + name
-		preProcessSoftwareExtraResults(fullQueryName, hostID, results, statuses, messages, query, logger)
+		preProcessSoftwareExtraResults(fullQueryName, host.ID, results, statuses, messages, query, logger)
+	}
+
+	// Filter out python packages that are also deb packages on ubuntu
+	pythonPackageFilter(host.Platform, results, statuses)
+}
+
+// pythonPackageFilter filters out duplicate python_packages that are installed under deb_packages on Ubuntu.
+// python_packages not matching a Debian package names are updated to "python3-packagename" to match OVAL definitions.
+func pythonPackageFilter(platform string, results *fleet.OsqueryDistributedQueryResults, statuses *map[string]fleet.OsqueryStatus) {
+	const pythonPrefix = "python3-"
```

Comment:
So, never `python` or `python2`?

Reply:
since python2 is EOL, we decided to omit.